test/Datasource.test.js

```diff
@@ -1,116 +1,190 @@
+      // emptyDataSource = new FileReadWriteDataSource(emptyFilename);
+    });
 
-        const emptyFilename = './fixtures/dataEmpty.json';    
-        emptyDataSource = new FileReadWriteDatasource(emptyFilename);
+  after(() => {
+    console.log('testing END');
+  });
+  
+  describe('#get', ()=> {
+    it('should return Promise ', ()=> {
+      expect(dataSource.get()).toBeA(Promise);
+    });
+
+    it('should resolve with null when key do not exist', ()=> {
+      return dataSource.get('test').then((res) => {
+        expect(res).toEqual(null);  
       });
-      after(() => {
-        console.log('testing END');
+    });
+
+    it('should resolve with value for existing key', ()=> {
+      return dataSource.get('city').then((res) => {
+        expect(res).toEqual('Kyiv');  
       });
-    
-    describe('#get', ()=> {
-        it('should return Promise ', ()=> {
-          expect(dataSource.get()).toBeA(Promise);
-        });
-    
-        it('should resolve with null when key do not exist', ()=> {
-          return dataSource.get('test').then((res) => {
-            expect(res).toEqual(null);  
-          });
+    });
+  })
+  
+  describe('#set', ()=> {
+    it('should return Promise ', ()=> {
+      expect(dataSource.set()).toBeA(Promise);
+    });
+
+    it('should resolve with seted key', ()=> {
+      return dataSource.set('key', 'value').then((res) => {
+        expect(res).toEqual('key');  
+      });
+    });
+  })
+  
+  describe('#delete', ()=> {
+    it('should return Promise ', ()=> {
+      expect(dataSource.delete()).toBeA(Promise);
+    });
+
+    it('should resolve with true', ()=> {
+      return dataSource.delete('key').then((res) => {
+        expect(res).toEqual(true);  
+      });
+    });
+  })
+
+  describe('chaining', () => {
+    it('should return Promise and resolve with a null', () => {
+      return emptyDataSource.get('key').then((res) => {
+        expect(emptyDataSource.get('key')).toBeA(Promise);
+        expect(res).toEqual(null);  
+      });
+    });
+
+    it('set>get shoud return with seted value', () => {
+      return emptyDataSource.set('key', 'value')
+        .then(() =>  emptyDataSource.get('key'))
+        .then((res) => {
+          expect(res).toEqual('value'); 
         });
-    
-        it('should resolve with value for existing key', ()=> {
-          return dataSource.get('city').then((res) => {
-            expect(res).toEqual('Kyiv');  
-          });
+    });
+
+    it('set>get>delete shoud return with true', () => {
+      return emptyDataSource.set('key', 'value')
+        .then(() =>  emptyDataSource.get('key'))
+        .then(() =>  emptyDataSource.delete('key'))
+        .then((res) => {
+          expect(res).toEqual(true); 
         });
-      })
-    
-      describe('#set', ()=> {
-        it('should return Promise ', ()=> {
-          expect(dataSource.set()).toBeA(Promise);
+    });
+
+    it('set>get>delete>get shoud return with null', () => {
```

Comment:
Just tell me where from this testcase should return and who is null and I'll be OK with this title.

Reply:
resolved